jrpcfs/middleware_test.go

```diff
@@ -697,6 +697,44 @@ func TestRpcGetContainerPrefixAndMarker(t *testing.T) {
 	assert.Equal(".git/logs/refs/stash", ents[3].Basename)
 }
 
+func TestRpcGetContainerPrefixAndDelimiter(t *testing.T) {
+	server := &Server{}
+	assert := assert.New(t)
+
+	request := GetContainerReq{
+		VirtPath:   testVerAccountName + "/" + "c-nested",
+		Marker:     "",
+		MaxEntries: 10000,
+		Prefix:     ".git/logs/refs/",
+		Delimiter:  "/",
+	}
+	response := GetContainerReply{}
+	err := server.RpcGetContainer(&request, &response)
+
+	assert.Nil(err)
+	assert.Equal(3, len(response.ContainerEntries))
+	ents := response.ContainerEntries
+	assert.Equal(".git/logs/refs/.DS_Store", ents[0].Basename)
+	assert.Equal(".git/logs/refs/heads", ents[1].Basename)
+	assert.Equal(".git/logs/refs/stash", ents[2].Basename)
+
+	// Try with a prefix without a trailing slash
+	request = GetContainerReq{
+		VirtPath:   testVerAccountName + "/" + "c-nested",
+		Marker:     "",
+		MaxEntries: 10000,
+		Prefix:     ".git/logs/refs",
+		Delimiter:  "/",
+	}
+	response = GetContainerReply{}
+	err = server.RpcGetContainer(&request, &response)
+
+	assert.Nil(err)
+	assert.Equal(1, len(response.ContainerEntries))
+	ents = response.ContainerEntries
+	assert.Equal(".git/logs/refs", ents[0].Basename)
+}
+
```

Comment:
Interesting... hadn't noticed this before - and please don't feel like you need to change... but the convention in Go test code is to reference your `t *testing.T` argument... and invoke APIs relative to that (see the description of what's available for `type T` at https://golang.org/pkg/testing/).

I take it all these `assert.*()` calls will simply panic. `go test` will definitely see this as a failed test...but in a less clean way I suppose. Alas, package test didn't deign to include all these assert functions you (and Sam) are using from package github.com/stretchr/testify/assert :-(.

Reply:
@edmc-ss The assert calls will show up as a failed test at the end of the run.  They will not panic.

Reply:
@edmc-ss The assert() calls don't panic.  They just fail the individual test.
After all tests are executed a summary of failures is printed.

Reply:
To be honest, I just copied and pasted one of Sam's tests and adapted it to my additions. What I know is I ran the tests, and they passed. In this case, yes, thanks to the line @edmc-ss pointed out.